### PR TITLE
Rename fixUnburnableNFToken to fixNonFungibleTokensV1_2

### PR DESCRIPTION
**Pull request by kennyzlei**
## High Level Overview of Change

Renames common amendment `fixUnburnableNFToken` to `fixNonFungibleTokensV1_2` to more accurately reflect bundling of changes and follow version convention of previous amendments to non-fungible tokens

### Context of Change

Renames common amendment, which includes the following changes in the feature branch:
- https://github.com/XRPLF/rippled/pull/4391
- https://github.com/XRPLF/rippled/pull/4346
- https://github.com/XRPLF/rippled/pull/4380
- https://github.com/XRPLF/rippled/pull/4403
- https://github.com/XRPLF/rippled/pull/4399

### Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Refactor (non-breaking change that only restructures code)
- [ ] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation Updates
- [ ] Release

## Future Tasks
- Merge feature branch `feature/nft-fixes` to `develop` in https://github.com/XRPLF/rippled/pull/4417

### CODE REVIEW

File: src/ripple/protocol/impl/Feature.cpp [hunk 1]
```diff
@@ -451,7 +451,7 @@ REGISTER_FIX    (fixTrustLinesToSelf,           Supported::yes, DefaultVote::no)
 REGISTER_FIX    (fixRemoveNFTokenAutoTrustLine, Supported::yes, DefaultVote::yes);
 REGISTER_FEATURE(ImmediateOfferKilled,          Supported::yes, DefaultVote::no);
 REGISTER_FEATURE(DisallowIncoming,              Supported::yes, DefaultVote::no);
-REGISTER_FIX    (fixUnburnableNFToken,          Supported::yes, DefaultVote::no);
+REGISTER_FEATURE(NonFungibleTokensV1_2,         Supported::yes, DefaultVote::no);
```
[the remark this answers is not in the document]
Comment (ledhed2222) [reply src/ripple/protocol/impl/Feature.cpp:454]: good question on this - i'm not sure either

[the remark this answers is not in the document]
Comment (intelliot) [reply src/ripple/protocol/impl/Feature.cpp:454]: For NonFungibleTokensV1_1, it was actually a feature as it actually implemented XLS-20 and was a replacement of NonFungibleTokensV1.

For NonFungibleTokensV1_2, it is actually a fix as it is just fixing bugs in the current XLS-20 implementation.

per some [community feedback here](https://github.com/XRPLF/rippled/pull/4399#issuecomment-1418599887), I believe we want to go with `fixNonFungibleTokensV1_2`.

As for the differences in naming in the const name in the header file - I don't know why that is. @seelabs @scottschurr ?

[the remark this answers is not in the document]
Comment (scottschurr) [reply src/ripple/protocol/impl/Feature.cpp:454]: @kennyzlei, the asymmetry you are seeing between the naming of `feature`s and `fix`es is caused by a difference between the `REGISTER_FEATURE` and `REGISTER_FIX` macros.  You'll find them here:

https://github.com/XRPLF/rippled/blob/develop/src/ripple/protocol/impl/Feature.cpp#L393-L406

You'll notice that the `REGISTER_FEATURE` macro contains `feature##fName`.  That has the effect of textually prepending the text `feature` to the supplied name.  The `REGISTER_FIX` macro doesn't have a construction like that.

What you _really_ want to know is why the macros were coded with that asymmetry.  I can only guess.  But my guess is that `feature` has more characters than `fix`.  Automatically prepending `feature` allowed the line for each feature creation to be shorter, so the lines were closer to our guidelines.  Doing the same for `fix` did not have as much impact on the line length, so it was left.  Just a working hypothesis.

At any rate, there's not much technical going on here.  Just different macro expansions.

File: src/ripple/app/tx/impl/NFTokenAcceptOffer.cpp [hunk 2]
```diff
@@ -121,8 +121,8 @@ NFTokenAcceptOffer::preclaim(PreclaimContext const& ctx)
         // If the buyer specified a destination
         if (auto const dest = bo->at(~sfDestination))
         {
-            // fixUnburnableNFToken
-            if (ctx.view.rules().enabled(fixUnburnableNFToken))
+            // fixNonFungibleTokensV1_2
```
Comment (intelliot) [inline src/ripple/app/tx/impl/NFTokenAcceptOffer.cpp:124]: what is the value of this comment?

File: src/ripple/app/tx/impl/NFTokenAcceptOffer.cpp [hunk 3]
```diff
@@ -139,8 +139,8 @@ NFTokenAcceptOffer::preclaim(PreclaimContext const& ctx)
         // If the seller specified a destination
         if (auto const dest = so->at(~sfDestination))
         {
-            // fixUnburnableNFToken
-            if (ctx.view.rules().enabled(fixUnburnableNFToken))
+            // fixNonFungibleTokensV1_2
```
Comment (intelliot) [inline src/ripple/app/tx/impl/NFTokenAcceptOffer.cpp:142]: same here (I wonder if this line could be removed)

**Review by intelliot — APPROVED** [review]
LGTM

File: src/ripple/app/tx/impl/NFTokenAcceptOffer.cpp (hunk 2 above)
Comment (kennyzlei) [reply src/ripple/app/tx/impl/NFTokenAcceptOffer.cpp:124]: yeah this does not seem to add more info in the comment beyond the code itself, ill remove

**Comment by kennyzlei** [comment]
closing in favor of https://github.com/XRPLF/rippled/pull/4416 which includes other cosmetic refactors

**Pull request closed without merging**